doc/source/overview/high_level.md

```diff
@@ -0,0 +1,508 @@
+
+$$
+\psi_j = \sum_{S\subset \{1,...,p\} \setminus \{j\}} \frac{|S|!(p - |S| - 1)!}{p!}(val(S \cup x_i) - val(S))
+$$
+
+The weights convey how much you can learn from a specific coalition. Large and Small coalitions mean more learned because we've isolated more of the effect. At the same time, medium size coalitions don't supply us with as much information because there are many possible such coalitions.
+
+The main issue with the above is that there will be many possible coalitions, $2^M$ to be precise. Hence instead of computing all of these, we use a sampling process on the space of coalitions and then estimate the Shapley values by training a linear model. Because a coalition is a set of players/features that are contributing to a prediction, we represent this as points in the space of binary codes $z' = \{z_0,...,z_m\}$ where $z_j = 1$ means that the $j^th$ feature is present in the coalition while $z_j = 0$ means it is not. To obtain the dataset on which we train this model, we first sample from this space of coalitions then compute the values of $f$ for each sample. We obtain weights for each sample using the Shapley Kernel:
+
+$$
+\pi_{x}(z') = \frac{M - 1}{\frac{M}{|z'|}|z'|(M - |z'|)}
+$$
+
+Once we have the data points, the values of $f$ for each data point, and the sample weights, we have everything we need to train a linear model. The paper shows that the coefficients of this linear model are the Shapley values.
+
+There is some nuance to how we compute the value of a model given a specific coalition, as most models aren't built to accept input with arbitrary missing values. If $D$ is the underlying distribution the samples are drawn from, then ideally, we would use the conditional expectation:
+
+$$
+f(S) = \mathbb{E}_{D}[f(x)|x_S]
+$$
+
+Computing this value is very difficult. Instead, we can approximate the above using the interventional conditional expectation, which is defined as:
+
+$$
+f(S) = \mathbb{E}_{D}[f(x)|do(x_S)]
+$$
+
+The $do$ operator here fixes the values of the features in $S$ and samples the remaining $\bar{S}$ feature values from the data. A Downside of interfering in the distribution like this can mean introducing unrealistic samples if there are dependencies between the features.
+
+**Pros**
+- The Shapley values are fairly distributed among the feature values
+- Shapley values can be easily interpreted and visualized
+- Very general as is a blackbox method
+
+**Cons**
+- KernalSHAP is slow owing to the number of samples required to estimate the Shapley values accurately
+- The interventional conditional probability introduces unrealistic data points
+- Requires access to the training dataset
+
+
+#### TreeSHAP
+
+| Model-types  | Task-types     | Data-types  |
+| ------------ | -------------- | ----------- |
+| Tree-based   | Classification | Tabular     |
```

Comment:
Let's list the supported libraries explicitly (consult our wrapper and the upstream `shap` library).